src/main/java/usi/si/seart/staticcodeanalysis/ClonedRepo.java

```diff
@@ -0,0 +1,41 @@
+package usi.si.seart.staticcodeanalysis;
+
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.experimental.FieldDefaults;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+
+/**
+ * A git repository which is cloned on the filesystem.
+ */
+@Slf4j
+@AllArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public
+class ClonedRepo implements AutoCloseable {
+    @Getter
+    Path path;
+
+    /**
+     * When used in a try-with-resources block, automatically deletes the folder on the filesystem once the clause is exited.
+     */
+    @Override
+    public void close() {
+        try {
+            Files.walk(path)
```

Comment:
I keep getting a warning here from Sonar: "Resources should be closed" [java:S2095](https://rules.sonarsource.com/java/RSPEC-2095)
Either address the warning or determine if FP